miniconf/tests/option.rs

```diff
@@ -118,7 +118,7 @@ fn option_test_defer_option() {
     s.set_json("/data", b"7").unwrap();
     assert_eq!(s.data, Some(7));
 
-    let mut iterator = S::iter_paths::<String>("/");
+    let mut iterator = S::iter_paths::<String>("/").count();
```

Comment:
Certainly testing purposes.
Also for the same reasons `Iterator` has `size_hint()` and `ExactSizeIterator`, i.e. correct (pre-)allocation for collections, and performance.
I wanted to make the default uncounted as we don't need it, it's a performance penalty, and because the `_unchecked` was inappropriate/confusing.